### ISSUE TITLE
False Positive UnnecessaryApply

### ISSUE THREAD

**Issue by scottkennedy** [arturbosch/detekt#1305]
## Expected Behavior
No `UnnecessaryApply` is reported

## Observed Behavior
`UnnecessaryApply` is reported

## Steps to Reproduce
```kt
val view = FrameLayout(context).apply {
    background = ColorDrawable(ContextCompat.getColor(context, R.color.white))
}
```

## Context
Since I'm using the return value, the `apply` isn't unnecessary.

## Your Environment
* Version of detekt used: 1.0.0-RC10
* Version of Gradle used (if applicable): 4.10.2
* Operating System and version: Various

**Comment by Tolriq** [comment]
Just tested that rule and got plenty of different false positives too.

```
                        .selectAll(MoviesTable.ALL_TABLE_COLUMNS)
                        .apply {
                            if (!genre.isNullOrEmpty()) {
                                where(MoviesTable.GENRES + " LIKE ?", "%$genre%")
                            }
                        }
                        .orderBy("RANDOM()")
```
or

```
    val databaseAdapter: DatabaseAdapter by lazy {
        DatabaseAdapter(this).apply {
            try {
                open()
            } catch (e: SQLiteException) {
                logError(TAG, e) { "Error with database" }
            }
        }
    }
```

**Comment by vanniktech** [comment]
Some more:

```kotlin
private val staticCirclePaint = Paint().apply { isAntiAlias = true }
fun setColor(color: Int) = apply { initialColor = color }
```

**Comment by zayo** [comment]
Another: 
```
private val requestedInterval by lazy {
    MutableLiveData<Int>().apply { value = UsageFragment.INTERVAL_DAY }
}
```

**Comment by patrick-iv-knowit** [comment]
I like the rule but it needs some additional constraints.
Having the same issue with
```kotlin
val itemAnimator = DefaultItemAnimator().apply { supportsChangeAnimations = false }
```

**Comment by realdadfish** [comment]
Yeah, I resided to replace `.apply` with `.also` in these places, because in these single cases I'm not too lazy to write the additional `it.` and wanted to keep notified about the real issue cases, but certainly, this is an arbitrary "workaround".

**Comment by mikezx6r** [comment]
I chose to suppress it and added a todo to follow up on the ticket. More work, though.

**Comment by cesards** [comment]
Happening to us as well.
```
override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
        inflater.inflate(productDetailsR.layout.fragment_product_details_tab, container, false).apply {
            containerView = this
        }
```

**Comment by cesards** [comment]
Is there any way we could change that rule? Do you think that this rule would make more sense when using `apply` and not using the result of it?

So detekt complains about:
```
questionHeaderSortByLabel.apply {
  contentDescription = resources.getString(R.string.ada_qna_sorted_by, sortedByValue)
}
```
but not about:
```
val content = Intent().apply { 
  putExtra("", 1)
}
```

**Comment by vanniktech** [comment]
This should be a blocker regarding 1.0.0
